include/zenoh-pico/protocol/core.h

```diff
@@ -227,6 +241,7 @@ typedef struct {
     _z_timestamp_t timestamp;
     _z_encoding_t encoding;
     z_sample_kind_t kind;
+    _z_qos_t qos;
```

Comment:
Pretty sure by putting the private type here, the public alias `z_qos_t` is never used. So I'm tempted to use `z_qos_t` instead.